packages/preferences/src/browser/folders-preferences-provider.ts

```diff
@@ -51,13 +51,13 @@ export class FoldersPreferencesProvider extends PreferenceProvider {
         Promise.all(readyPromises).then(() => this._ready.resolve());
     }
 
-    protected updateProviders(): void {
+    protected async updateProviders(): Promise<void> {
         const roots = this.workspaceService.tryGetRoots();
         const toDelete = new Set(this.providers.keys());
         for (const folder of roots) {
-            for (const configPath of this.configurations.getPaths()) {
+            for (const configPath of await this.configurations.getPaths()) {
```

Comment:
This has to be sync to initialize default preferences properly. If it is async a client can call get method and receive  undefined instead of default value.

Reply:
Then we have to keep `PreferenceConfigurations` sync which means we have to have `EnvVariablesServerImpl` sync.

Reply:
It is not possible on the frontend, but you can do https://github.com/eclipse-theia/theia/pull/6650#discussion_r358754429, i.e. keep `getPaths` sync, but do async fetching with inversify.js async provider